lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc

```diff
@@ -32,3 +37,23 @@
        ->execute();
    }
  }
+
+ function dosomething_gladiator_retry_failed_users() {
+   $task_log = db_select('dosomething_gladiator_failed_task_log', 't')
+    ->fields('t')
+    ->execute()
+    ->fetchAll();
+
+   foreach ($task_log as $task) {
+      $user = user_load($task->uid);
+      $values = [
+        'nid' => $task->nid,
+        'run_nid' => $task->run_nid,
+      ];
+      db_delete('dosomething_gladiator_failed_task_log')
```

Comment:
it can be done right after you fetch all rows from the table, not after the loop